### DISCUSSION TITLE
Why client retryTimeout always be 1?

### DISCUSSION THREAD

**Discussion by ray2011** [redis/go-redis#2069] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7893714/162110695-261fe5d8-7923-47de-9028-bf01a5023e33.png)
Is this safe for all redis commands?

**Comment by vmihailenco** [comment, marked as answer]
Nope, this looks like bug. According to [Git history](https://github.com/go-redis/redis/commit/02ccf05ed073709d0c4a3b933a50b13a2b87c0b6), it should be:

```go
if cmd.readTimeout() == nil {
	atomic.StoreUint32(&retryTimeout, 1)
} else {
	atomic.StoreUint32(&retryTimeout, 0)
}
```

Does that make sense?

**Comment by ray2011** [comment]
@vmihailenco May I pull a request to fix it?

**Reply by vmihailenco** [reply]
Yes, thanks
